src/matrixrtc/MatrixRTCSession.ts

```diff
@@ -825,7 +825,8 @@ export class MatrixRTCSession extends TypedEventEmitter<MatrixRTCSessionEvent, M
 
         const myCallMemberEvent = roomState.getStateEvents(EventType.GroupCallMemberPrefix, localUserId) ?? undefined;
         const content = myCallMemberEvent?.getContent() ?? {};
-        const legacy = "memberships" in content || this.useLegacyMemberEvents;
+        const legacy =
+            (("memberships" in content && content.memberships?.length) ?? 0 > 0) || this.useLegacyMemberEvents;
```

Comment:
This will fallback to the legacy format only if the _current user's_ `*.call.member` state event uses the legacy format.  Should it fallback if _any_ user's `*.call.member` state uses the legacy format?

Reply:
I was thinking about this as well. But then decided that we can make the transition a bit more abrupt and only added this usecase to save time. I saw your PR however and like it more then this. So I will remove this change from this pr.